userspace/libsinsp/container_engine/cri.cpp

```diff
@@ -193,17 +193,24 @@ void cri::set_extra_queries(bool extra_queries) {
 bool cri::resolve(sinsp_container_manager* manager, sinsp_threadinfo* tinfo, bool query_os_for_missing_info)
 {
 	sinsp_container_info container_info;
+	sinsp_container_info *existing_container_info;
 
 	if(matches_runc_cgroups(tinfo, CRI_CGROUP_LAYOUT, container_info.m_id))
 	{
-		container_info.m_type = s_cri_runtime_type;
+		// It might also be a docker container, so set the type
+		// to UNKNOWN for now.
+		container_info.m_type = CT_UNKNOWN;
```

Comment:
What happens when both docker and cri fail to return metadata for a container? Are we stuck with CT_UNKNOWN?

Reply:
Yes. I suppose we could have a combined type or just pick one of them, but I think it'd be better to have this explicit type for when we don't know the metadata.

Reply:
@gnosek and I chatted offline and we're going to get rid of CT_UNKNOWN and rely on m_metadata_complete to internally denote that container info is incomplete. Since docker runs first, it will set the type to CT_DOCKER. If cri looks up info successfully it will set the type properly. If both fail, the type will remain at CT_DOCKER, which seems as good as you can given the cgroups are the same.